sentry-beam/src/main/java/io/sentry/beam/SentryBeam.java

```diff
@@ -0,0 +1,192 @@
+package io.sentry.beam;
+
+import io.sentry.Sentry;
+import io.sentry.event.Event;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.PipelineRunner;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.transforms.windowing.BoundedWindow;
+import org.apache.beam.sdk.transforms.windowing.PaneInfo;
+import org.joda.time.Instant;
+
+public final class SentryBeam  {
+    /**
+     * Hide constructor
+     */
+    private SentryBeam() {
+
+    }
+
+    public static void capture(Event event) {
```

Comment:
Do we want to expose a `capture` method here?
This might be confusing for docs etc when we have multiple entry points to the API.
Perhaps the way to go could be a client factory.

Like: https://github.com/getsentry/sentry-java/blob/master/sentry-appengine/src/main/java/io/sentry/appengine/AppEngineSentryClientFactory.java#L36

Reply:
I had initially started with a client factory, see https://github.com/getsentry/sentry-java/pull/741/files/2ec5244751560bbc484cebe048db84aeabcb203c. However that my issue what the client factory was that it required me to pass it the variables like `timestamp`, `boundedWindow`, etc. at initialization time. For example, the end user code would look like

```
SentryClientFactory clientFactory = BeamSentryClientFactory.Builder.withTimestamp(timestamp)
        .withBoundedWindow(boundedWindow)
        // more
        .build();
Sentry.init("your dsn", clientFactory);

try {
    // beam code
} catch (Exception e) {
    Sentry.capture(e);
}
```
vs.
```
Sentry.init("your dsn");

try {
   // beam code
} catch (Exception e) {
    SentryBeam.withTimestamp(timestamp)
             .withBoundedWindow(boundedWindow)
             // more
             .capture(e);
}
```
I thought this implementation makes more sense as the first attempt makes it so that every event would have the same tags attached. Which might be okay in some distributed cases, but when running locally, it would overwrite the original sentry client and attach the wrong information onto the event. If there was a way to guarantee it's using a different sentry client, that may be better.